src/browser/chrome.ts

```diff
@@ -40,16 +40,49 @@ export const closeBrowser = async () => {
 };
 
 const WINDOW_VARIABLE_NAME = `___webrender`;
+const LAST_RESPONSE_AT_VARIABLE_NAME = `__webrenderLastResponseAt`;
+const COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME = `__webrenderCounterReqUnfinished`;
 export const getPageInitScriptFor = (
   js: string,
-  jsOn: JsOn = "commit"
-) => `try {
-  const AsyncFunction = Object.getPrototypeOf(
+  jsOn: JsOn = "commit",
+  timeout: number
+) => {
+  const PAGE_TRUE_LOAD = `
+  async pageTrueLoad ({
+    idleTimeout = 2000, resolveBeforeRenderingTimeout = 2000
+  } = {}) {
+    const timeoutPromise = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
```

Comment:
I suggest to return something like `{ ok: boolean, timeout: boolean }` indicating whether we have the exit because of the timeout or a "clean exit"

Reply:
What does `ok` mean and how is `ok` different from `timeout`? Maybe we should name it `isTimeouted` or `timeouted`?

Reply:
`isTimeouted` is fine too, perhaps `isTimeout` will sound better. `ok` serves the purpose of "generic ok", because maybe later the error might not only be timeout